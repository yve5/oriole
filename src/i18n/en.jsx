const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
