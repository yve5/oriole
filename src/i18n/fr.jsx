const I18N_DICTIONARY = {
  LANG: 'fr',
  LANGUAGE: 'Français',
  DATE_FORMAT: 'dd/MM/yyyy',
  LOCALE: 'fr',
  '$number items per page': '$number éléments par page',
  '$total records': '$total enregistrements',
  Add: 'Ajouter',
  'Ascending order': 'Ordre croissant',
  Back: 'Retour',
  'Changes have been saved.': 'Les changements ont été sauvegardés.',
  'Choose a file': 'Choisir un fichier',
  Columns: 'Colonnes',
  'Creation date': 'Date de création',
  'Descending order': 'Ordre décroissant',
  Disabled: 'Désactivé',
  Enabled: 'Activé',
  First: 'Premier',
  'Hello World': 'Bonjour le monde',
  'Hide columns': 'Cacher les colonnes',
  Last: 'Dernier',
  'Loading...': 'Chargement...',
  Members: 'Membres',
  Next: 'Suivant',
  No: 'Non',
  'No record': 'Aucun enregistrement',
  'One record': 'Un enregistrement',
  Previous: 'Précédent',
  Reset: 'Réinitialiser',
  'This field is invalid.': 'Ce champ est invalide.',
  Title: 'Titre',
  'to fill': 'à remplir',
  'Update of $update, created at $start by $guest':
    'Mise à jour le $update, créé le $start par $guest',
  Yes: 'Oui',
  Changelog: 'Journal des modifications',
  'About us': 'À propos de nous',
  Collapse: 'Réduire',
  Expand: 'Développer',
  Paging: 'Pagination',
  Buttons: 'Boutons',
  'Something really long': 'Quelque chose réellement long',
};

export default I18N_DICTIONARY;
