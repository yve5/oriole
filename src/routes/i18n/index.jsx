import I18nPage from './components/I18n';

export default I18nPage;
