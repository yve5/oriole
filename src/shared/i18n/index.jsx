import LangSwitch from './components/LangSwitch';
import I18n from './components/I18n';
import i18nReducer from './reducer';

export { I18n, LangSwitch, i18nReducer };
