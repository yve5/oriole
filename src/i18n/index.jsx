import fr from './fr';
import en from './en';
import { tsl } from '../shared/i18n/utils/Translate';

const i18n = {
  tsl: (term, args) => tsl(fr, term, args),
  dictionaries: [fr, en],
  lang: fr.LANG,
};

export default i18n;
