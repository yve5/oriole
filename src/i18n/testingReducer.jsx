import i18n from '../shared/i18n/reducer';
import fr from './fr';
import en from './en';

export default i18n([fr, en])();
