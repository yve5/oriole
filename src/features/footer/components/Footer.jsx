import React from 'react';
import { clsx } from 'clsx';
import { I18n } from '../../../shared/i18n';

const Footer = () => (
  <footer
    className={clsx(
      'd-flex',
      'flex-wrap',
      'justify-content-around',
      'align-items-center',
      'py-2',
      'border-top'
    )}
  >
    <p className={clsx('col-md-4', 'mb-0', 'p-2', 'text-body-secondary')}>
      Â© Oriole, 2024
    </p>

    <ul className={clsx('nav', 'col-md-4', 'justify-content-end')}>
      <li className={clsx('nav-item')}>
        <button
          type="button"
          className={clsx('nav-link', 'p-2', 'text-body-secondary')}
        >
          <I18n>About us</I18n>
        </button>
      </li>
      <li className={clsx('nav-item')}>
        <button
          type="button"
          className={clsx('nav-link', 'p-2', 'text-body-secondary')}
        >
          <I18n>Changelog</I18n>
        </button>
      </li>
    </ul>
  </footer>
);

export default Footer;
