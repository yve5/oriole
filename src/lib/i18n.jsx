import { I18n, LangSwitch, i18nReducer } from '../shared/i18n';

export { I18n, LangSwitch, i18nReducer };
