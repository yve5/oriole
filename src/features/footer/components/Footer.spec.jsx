import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import Footer from './Footer';
import i18n from '../../../i18n';

const mockStore = configureMockStore([thunk]);

describe('Footer', () => {
  it('renders the component. default', () => {
    const { asFragment } = render(
      <Provider store={mockStore({ i18n })}>
        <MemoryRouter>
          <Footer />
        </MemoryRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
