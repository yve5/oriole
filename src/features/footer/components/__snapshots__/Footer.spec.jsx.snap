// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > renders the component. default 1`] = `
<DocumentFragment>
  <footer
    class="d-flex flex-wrap justify-content-around align-items-center py-2 border-top"
  >
    <p
      class="col-md-4 mb-0 p-2 text-body-secondary"
    >
      © Oriole, 2024
    </p>
    <ul
      class="nav col-md-4 justify-content-end"
    >
      <li
        class="nav-item"
      >
        <button
          class="nav-link p-2 text-body-secondary"
          type="button"
        >
          À propos de nous
        </button>
      </li>
      <li
        class="nav-item"
      >
        <button
          class="nav-link p-2 text-body-secondary"
          type="button"
        >
          Journal des modifications
        </button>
      </li>
    </ul>
  </footer>
</DocumentFragment>
`;
